src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -208,7 +208,7 @@ public final Connection getConnection(final long hardTimeout) throws SQLExceptio
       if (originalException instanceof SQLException) {
          sqlState = ((SQLException) originalException).getSQLState();
       }
-      throw new SQLTimeoutException(String.format("Timeout after %dms of waiting for a connection.", clockSource.elapsedMillis(startTime)), sqlState, originalException);
+      throw new SQLTimeoutException(poolName + " - Cannot acquire connection, Timeout after " + clockSource.elapsedMillis(startTime) + "ms.", sqlState, originalException);
```

Comment:
While we are touching this code, it turns out that [SQLTimeoutException](http://docs.oracle.com/javase/7/docs/api/java/sql/SQLTimeoutException.html) is not the correct exception to throw here.  HikariCP should be throwing [SQLTransientConnectionException](http://docs.oracle.com/javase/7/docs/api/java/sql/SQLTransientConnectionException.html) instead.  Both are subclasses of ``SQLTransientException``, but ``SQLTimeoutException`` is only for *Statement* execution timeouts.